interhost.py

```diff
@@ -262,6 +270,94 @@ def main_align_mafft(args):
+        help='Location for the output files (default is cwd: %(default)s)')
+    parser.add_argument('--outFilePrefix', default="singlechr",
+        help='Prefix for the output file name (default: %(default)s)')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--localpair', default=None, action='store_true',
+        help='All pairwise alignments are computed with the Smith-Waterman algorithm.')
+    group.add_argument('--globalpair', default=None, action='store_true',
+        help='All pairwise alignments are computed with the Needleman-Wunsch algorithm.')
+    parser.add_argument('--preservecase', default=None, action='store_true',
+        help='Preserve base or aa case, as well as symbols.')
+    parser.add_argument('--reorder', default=None, action='store_true',
+        help='Output is ordered aligned rather than in the order of the input (default: %(default)s).')
+    parser.add_argument('--gapOpeningPenalty', default=1.53, type=float,
+        help='Gap opening penalty (default: %(default)s).')
+    parser.add_argument('--ep', type=float,
+        help='Offset (works like gap extension penalty).')
+    parser.add_argument('--verbose', default=False, action='store_true',
+        help='Full output (default: %(default)s).')
+    parser.add_argument('--outputAsClustal', default=None, action='store_true',
+        help='Write output file in Clustal format rather than FASTA')
+    parser.add_argument('--maxiters', default = 0, type=int,
+        help='Maximum number of refinement iterations (default: %(default)s). Note: if "--localpair" or "--globalpair" is specified this defaults to 1000.')
+    parser.add_argument('--threads', default = -1, type=int,
+        help='Number of processing threads (default: %(default)s, where -1 indicates use of all available cores).')
+    util.cmd.common_args(parser, (('loglevel', None), ('version', None), ('tmpDir', None)))
+    util.cmd.attach_main(parser, multichr_mafft)
+    return parser
+
+def multichr_mafft(args):
+    ''' Run the mafft alignment on a series of chromosomes provided in sample-partitioned FASTA files. Output as FASTA.
+        (i.e. file1.fasta would contain chr1, chr2, chr3; file2.fasta would also contain chr1, chr2, chr3)'''
+
+    if int(args.threads) == 0 or int(args.threads) < -1:
+        raise argparse.ArgumentTypeError('Argument "--threads" must be non-zero. Specify "-1" to use all available cores.')
+
+    # get the absolute path to the output directory in case it has been specified as a relative path,
+    # since MAFFT relies on its CWD for path resolution
+    absoluteOutDirectory = os.path.abspath(args.outDirectory)
+
+    # prefix for output files
+    prefix = "" if args.outFilePrefix == None else args.outFilePrefix
+
+    # reorder the data into new FASTA files, where each FASTA file has only variants of its respective chromosome
+    transposedFiles = transposeChromosomeFiles(args.inFastas, absoluteOutDirectory, prefix)
```

Comment:
I don't like the temp files going into the absoluteOutDirectory.  I also don't particularly see why the temp files need to have human-readable names.  Perhaps have transposeChromosomeFiles just spit out a bunch of files that were created with util.file.mkstempfname and then move the code that creates the final output filename string into the for loop right below here?

Then the post-cleanup code below becomes much simpler, and even optional (since the util.cmd framework makes a point of blowing away the tempfiles after exit anyway).